types/types.ts

```diff
@@ -70,36 +70,64 @@ export enum ProposalCategory {
   None
 }
 
-export type ProposalConfig = {
+export interface ProposalConfig {
   deploy: boolean;
   category: ProposalCategory;
   totalValue: number;
   proposal: ProposalDescription;
   affectedContractSignoff: string[];
   deprecatedContractSignoff: string[];
   proposalId: string;
-};
+}
+
+export interface TemplatedProposalConfig {
+  deploy: boolean;
+  category: ProposalCategory;
+  totalValue: number;
+  proposal: TemplatedProposalDescription;
+  affectedContractSignoff: string[];
+  deprecatedContractSignoff: string[];
+  proposalId: string;
+}
 
-export type ProposalsConfigMap = {
+export interface ProposalsConfigMap {
   [key: string]: ProposalConfig;
-};
+}
+
+export interface TemplatedProposalsConfigMap {
+  [key: string]: TemplatedProposalConfig;
+}
 
-export type ProposalDescription = {
+export interface ProposalDescription {
   title: string;
   commands: ProposalCommand[];
   description: string;
-};
+}
 
-export type ProposalCommand = {
+export interface TemplatedProposalDescription {
```

Comment:
type --> interface because you should always use interfaces over types where sensible (they're full objects, not just aliases, and are thus capable of introspection)

added in the templatedproposal stuff in addition to the normal proposal stuff, to enable the replacement for replaceArgs